python/src/aiconfig/schema.py

```diff
@@ -444,30 +445,134 @@ def get_model_metadata(
     # `update_model_parser_registry_with_config_runtime`` function
     # Tracked in https://github.com/lastmile-ai/aiconfig/issues/503
     def update_model(
-        self, model_metadata: Dict | ModelMetadata, prompt_name: Optional[str] = None
+        self, name: Optional[str] = None, settings: Optional[InferenceSettings] = None, prompt_name: Optional[str] = None
     ):
         """
-        Updates model settings in AIconfig-level metadata
+        Updates model name and/or settings at the prompt (if specified) or AIConfig level.
 
         Args:
-            model_metadata (dict): The model metadata to update.
-            prompt_name (str, optional): If specified, the model settings will only be applied to the prompt with the given prompt_name.
+            name (str): The model name to update. If None, keep existing name.
+            settings (dict): The model settings to update. If None, keep existing settings.
+            prompt_name (Optional[str]): If specified, the model updatd will only be applied to the prompt with the given prompt_name.
```

Comment:
nit: Consider adding a couple of examples in the docstring to illustrate the method's various use cases.